src/plots/cartesian/axes.js

```diff
@@ -2946,20 +2946,42 @@ function calcLabelLevelBbox(ax, cls, mainLinePositionShift) {
         bottom = -Infinity;
         left = Infinity;
         right = -Infinity;
+        var xAxis = ax._id.charAt(0) === 'x';
         ax._selections[cls].each(function() {
             var thisLabel = selectTickLabel(this);
+            var hidden = thisLabel.style('display') === 'none';
             // Use parent node <g.(x|y)tick>, to make Drawing.bBox
             // retrieve a bbox computed with transform info
             //
             // To improve perf, it would be nice to use `thisLabel.node()`
             // (like in fixLabelOverlaps) instead and use Axes.getPxPosition
             // together with the makeLabelFns outputs and `tickangle`
             // to compute one bbox per (tick value x tick style)
+            if (hidden) {
+                // turn on label display temporarily to calculate its bbox
+                thisLabel.style('display', null);
+            }
             var bb = Drawing.bBox(thisLabel.node().parentNode);
-            top = Math.min(top, bb.top);
-            bottom = Math.max(bottom, bb.bottom);
-            left = Math.min(left, bb.left);
-            right = Math.max(right, bb.right);
+            if (hidden) {
+                selectTickLabel(this).style('display', 'none');
```

Comment:
for consistency

(unless there's a good reason not to do this)

```suggestion
                thisLabel.style('display', 'none');
```

Reply:
Never mind, I didn't really believe that this was the correct fix, so I guess I was a little bit sloppy. The code doesn't exist anymore.